Move styles to head element

jsteele883 opened the pull request:
- Rendering the un-mergable rules in <head></head of the html rather than within the <body></body> will allow for responsiveness on gmail apps and best practice email output.

avit commented on lib/premailer/adapter/nokogiri.rb:
```diff
@@ -150,9 +150,9 @@ def write_unmergable_css_rules(doc, unmergable_rules) # :nodoc:
           unless (body = doc.search('body')).empty?
             if doc.at_css('body').children && !doc.at_css('body').children.empty?
-              doc.at_css('body').children.before(::Nokogiri::XML.fragment(style_tag))
+              doc.at_css('head').children.after(::Nokogiri::XML.fragment(style_tag))
```
These conditions are conflicting: "if there's a body, put it in the head".

If anything, this should be checking for presence of `<head>` and add it if needed.

avit commented on lib/premailer/adapter/nokogiri.rb:
```diff
@@ -150,9 +150,9 @@ def write_unmergable_css_rules(doc, unmergable_rules) # :nodoc:
+              doc.at_css('head').children.after(::Nokogiri::XML.fragment(style_tag))
             else
-              doc.at_css('body').add_child(::Nokogiri::XML.fragment(style_tag))
+              doc.at_css('body').add_next_sibling(::Nokogiri::XML.fragment(style_tag))
```
This will insert the style node after body (outside of it). I don't think this would be recognized any more than having it inside the body: since `<head><style>` is recognized everywhere now, is it even needed?

jsteele883 committed:
Move styles to the head element

Updates to Gmail mobile apps mean that styles placed inside the <body> tag will be stripped. It is better supported all round to have style element within <head> tags and outside of the body. The following article outlines recent rendering changes:
https://www.campaignmonitor.com/blog/email-marketing/2016/10/gmail-update-googles-rendering-refresh/

jsteele883 committed:
syntax correction for #337
